### PR TITLE
Rwa 1006 task initiation 2

### PR DESCRIPTION
**Pull request by serdaroguzhanakin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-1006

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by mkayad — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/clients/CamundaClient.java [hunk 1]
```diff
@@ -78,5 +80,25 @@ String deleteProcessInstance(@RequestHeader(SERVICE_AUTHORIZATION) String servic
...
+
+    @GetMapping(value = "/process-instance/{id}/variables",
+        consumes = APPLICATION_JSON_VALUE,
+        produces = APPLICATION_JSON_VALUE
+    )
+    @ResponseBody
+    Map<String, ProcessVariable> getProcessInstanceVariablesById(
+        @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthorisation,
+        @RequestParam("id") String id
+    );
```
Comment (indvamshi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/clients/CamundaClient.java:102]: Are these changes required , can't it written in RestApiActions class?

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/clients/model/ProcessVariable.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public Object getValue() {
+        return value;
+    }
+
+    public String getType() {
+        return type;
+    }
+}
```
Comment (indvamshi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/clients/model/ProcessVariable.java:30]: Are these changes required?

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/clients/model/ProcessVariables.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    public Boolean getEnded() {
+        return ended;
+    }
+
+    public String getTenantId() {
+        return tenantId;
+    }
+
+}
```
Comment (indvamshi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/clients/model/ProcessVariables.java:67]: Are these changes required?

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/DocumentManagementFiles.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+            String contentType;
+
+            if (filename.endsWith(".PDF")) {
+                contentType = "application/pdf";
+
+            } else if (filename.endsWith(".DOC")) {
+                contentType = "application/msword";
+
+            } else if (filename.endsWith(".DOCX")) {
```
Comment (indvamshi) [inline src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/DocumentManagementFiles.java:63]: literal should be checked with value

**Review by indvamshi — CHANGES_REQUESTED** [review]

**Review by indvamshi — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/wataskmonitor/controllers/MonitorTaskJobControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Test
+    public void should_succeed_and_return_job_name() {
+        MonitorTaskJobRequest monitorTaskJobRequest = new MonitorTaskJobRequest(new JobDetails(JobName.CONFIGURATION));
+
+        MonitorTaskJobRequest response = monitorTaskJobController.monitorTaskJob(monitorTaskJobRequest);
+        assertEquals(JobName.CONFIGURATION, response.getJobDetails().getName());
+        verify(monitorTaskJobService, times(1)).execute(monitorTaskJobRequest.getJobDetails().getName());
+    }
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmonitor/controllers/MonitorTaskJobControllerTest.java:37]: do we need to also include a test to cover if the jobRequest fails?

Comment (serdaroguzhanakin) [reply src/test/java/uk/gov/hmcts/reform/wataskmonitor/controllers/MonitorTaskJobControllerTest.java:37]: there no exception thrown in code.

File: src/test/java/uk/gov/hmcts/reform/wataskmonitor/controllers/MonitorTaskJobControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    private MonitorTaskJobController monitorTaskJobController;
+
+    @BeforeEach
+    public void setup() {
+        monitorTaskJobController = new MonitorTaskJobController(monitorTaskJobService);
+    }
+
+    @Test
+    public void should_succeed_and_return_job_name() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmonitor/controllers/MonitorTaskJobControllerTest.java:31]: can remove public here as we are using Junit 5

File: src/test/java/uk/gov/hmcts/reform/wataskmonitor/services/controllers/MonitorTaskJobServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+        jobServices = List.of(initiationJob);
+
+        monitorTaskJobService = new MonitorTaskJobService(
+            jobServices, authTokenGenerator
+        );
+    }
+
+    @Test
+    public void should_run_initiation_job() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmonitor/services/controllers/MonitorTaskJobServiceTest.java:45]: an remove public here as we are using Junit 5

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+            userInfo.getUid(),
+            "IA",
+            "Asylum",
+            true,
+            caseDataContent
+        );
+
+        log.info("Created case [" + caseDetails.getId() + "]");
+
+        StartEventResponse submitCase = coreCaseDataApi.startEventForCaseWorker(
```
Comment (BenNzewi) [inline src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java:129]: Pact test for startEventForCaseWorker. See https://github.com/hmcts/wa-task-management-api/pull/249/files as you can reuse these.

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+            .eventToken(submitCase.getToken())
+            .event(Event.builder()
+                .id(submitCase.getEventId())
+                .summary("summary")
+                .description("description")
+                .build())
+            .data(data)
+            .build();
+
+        coreCaseDataApi.submitEventForCaseWorker(
```
Comment (BenNzewi) [inline src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java:149]: Pact test for submitEventForCaseWorker. See https://github.com/hmcts/wa-task-management-api/pull/249/files as you can reuse these.

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+
+    public String createCcdCase() {
+        Headers headers = authorizationHeadersProvider.getLawFirmAuthorization();
+        String userToken = headers.getValue(AUTHORIZATION);
+        String serviceToken = headers.getValue(SERVICE_AUTHORIZATION);
+        UserInfo userInfo = authorizationHeadersProvider.getUserInfo(userToken);
+
+        Document document = documentManagementFiles.getDocument(NOTICE_OF_APPEAL_PDF);
+
+        StartEventResponse startCase = coreCaseDataApi.startForCaseworker(
```
Comment (BenNzewi) [inline src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java:73]: Pact test for startForCaseworker. See https://github.com/hmcts/wa-task-management-api/pull/249/files as you can reuse these.

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+            .event(Event.builder()
+                .id(startCase.getEventId())
+                .summary("summary")
+                .description("description")
+                .build())
+            .data(data)
+            .build();
+
+        //Fire submit event
+        CaseDetails caseDetails = coreCaseDataApi.submitForCaseworker(
```
Comment (BenNzewi) [inline src/testUtils/java/uk/gov/hmcts/reform/wataskmonitor/config/GivensBuilder.java:117]: Pact test for submitForCaseworker. See https://github.com/hmcts/wa-task-management-api/pull/249/files as you can reuse these.

**Review by BenNzewi — APPROVED** [review]
